tracr/compiler/craft_graph_to_model.py

```diff
@@ -131,15 +120,15 @@ def _allocate_modules_to_layers(graph: nx.DiGraph,
   layer_allocation: Dict[int, int] = collections.defaultdict(lambda: -1)
   depth_by_node_id: Dict[int, int] = dict()
   nodes_by_depth: Dict[int, List[Node]] = collections.defaultdict(list)
-
-  # Compute depth of all model components (longest path from source to node)
+  computational_depth = compute_computational_depth(graph, [src[nodes.ID] for src in sources])
```

Comment:
I think this could directly be assigned to `depth_by_node_id`. Shouldn't matter that there are maybe more entries

Reply:
It fails tests if keep the extra entries:
```
FAIL: test_allocate_modules_to_layers_returns_expected_result (__main__.CraftAllocateModulesToLayersTest)
CraftAllocateModulesToLayersTest.test_allocate_modules_to_layers_returns_expected_result
Creates a graph and checks if the correct layer assignment is returned.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/WSL/working/itracr/inverse-tracr/tracr_module/tracr/compiler/craft_graph_to_model_test.py", line 152, in test_allocate_modules_to_layers_returns_expected_result
    self.assertEqual(layer_allocation[node_id], node["expected_result"])
AssertionError: 10 != -1

======================================================================
FAIL: test_allocate_modules_to_layers_returns_expected_result_for_chain (__main__.CraftAllocateModulesToLayersTest)
CraftAllocateModulesToLayersTest.test_allocate_modules_to_layers_returns_expected_result_for_chain
Tests a chain of alternating attention layers and MLPs.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/d/WSL/working/itracr/inverse-tracr/tracr_module/tracr/compiler/craft_graph_to_model_test.py", line 189, in test_allocate_modules_to_layers_returns_expected_result_for_chain
    self.assertEqual(layer_allocation[node_id], node["expected_result"])
AssertionError: 8 != -1

----------------------------------------------------------------------
Ran 3 tests in 0.009s
```

Reply:
Ah we do iterate over depth_by_node_id further down. Sorry missed that, your solution is good then!